pkg/cli/admin/catalog/mirror.go

```diff
@@ -559,7 +561,12 @@ func (o *MirrorCatalogOptions) Run() error {
 	}
 	mapping, err := indexMirrorer.Mirror()
 	if err != nil {
-		fmt.Fprintf(o.IOStreams.ErrOut, "errors during mirroring. the full contents of the catalog may not have been mirrored: %s\n", err.Error())
+		err = fmt.Errorf("errors during mirroring. the full contents of the catalog may not have been mirrored: %w", err)
+		if !o.ContinueOnError {
+			return err
+		} else {
+			fmt.Fprintln(o.IOStreams.ErrOut, err.Error())
+		}
```

Comment:
nit: this can be consolidated into a single condition:

```go
if !o.ContinueOnError {
    return err
}
fmt.Fprintln(o.IOStreams.ErrOut, err.Error())
```